### PR TITLE
Optimizations

### PR DESCRIPTION
**Pull request by ik-make**
Hey @jlandon 👋🏻 Here is most tricky and complex PR concerning the general update of epub reader dependencies 📖
I made "impossible" merge and it seems that it even work after 😅 
As I understand we have two main additional features: 

- loading handling 
- text scale 

Am I right or I missed something that I also should check? Html example works well, [ereader](https://github.com/ovenbits/messenger-ereader-flutter) example also works well. 

Hope I didn't miss anything 🤞🏻 Looking forward for your review  

mission impossible

### CODE REVIEW

**Review by jlandon — COMMENTED** [review]
@ik-make next week, can you check on the unit tests? I ran the unit tests in the project (`flutter test`) and 16 failing tests were reported. But when I cloned and tested the [original repo](https://github.com/Sub6Resources/flutter_html), all the tests passed 🤔

It might be nothing, but it would be good to investigate and confirm before merging 🕵️‍♂️

**Comment by ik-make** [comment]
@jlandon great call! 👍🏻 Just checked - unit tests were failed because of async parsing tree logic (for loading state feature) After I added delay before `expect` calls everything started working fine 🎉 
Also I've made double check all merge things, and made some small code optimizations during second review. 
Please let me know if you still see any issues 🙏🏻

**Review by jlandon — APPROVED** [review]
@ik-make looks good! Thanks for making those fixes 💯

**Pull request merged**
